salt/metalk8s/volumes/prepared/init.sls

```diff
@@ -3,51 +3,78 @@ include:
 
 {%- set volumes_to_create = [] %}
 {%- set all_volumes = pillar.metalk8s.volumes %}
-{%- set target_volume = pillar.get('volume') %}
+{%- set target_volume_name = pillar.get('volume') %}
 
 {#- If a volume is given we only take this one, otherwise we take them all. #}
-{%- if target_volume is not none %}
-  {%- if target_volume in all_volumes.keys() %}
+{%- if target_volume_name is not none %}
+  {%- set target_volume = all_volumes.get(target_volume_name) %}
+  {%- if target_volume is not none %}
     {%- do volumes_to_create.append(target_volume) %}
   {%- else %}
-Volume {{ target_volume }} not found in pillar:
+Volume {{ target_volume_name }} not found in pillar:
   test.configurable_test_state:
-    - name: {{ target_volume }}
+    - name: {{ target_volume_name }}
     - changes: False
     - result: False
-    - comment: Volume {{ target_volume }} not found in pillar
+    - comment: Volume {{ target_volume_name }} not found in pillar
   {%- endif %}
 {%- else %}
-    {%- do volumes_to_create.extend(all_volumes.keys()|list) %}
+    {%- do volumes_to_create.extend(all_volumes.values()|list) %}
 {%- endif %}
 
 Create the sparse file directory:
   file.directory:
     - name: /var/lib/metalk8s/storage/sparse/
     - makedirs: True
 
+Set up systemd template unit for sparse loop device provisioning:
+  file.managed:
+    - name: /etc/systemd/system/metalk8s-sparse-volume@.service
+    - source: salt://{{ slspath }}/files/metalk8s-sparse-volume@.service
+    - user: root
+    - group : root
+    - mode: 644
+    - makedirs: True
```

Comment:
No. If `/etc/systemd/system` doesn't exist then something is *really* wrong on the system and we shouldn't ignore that.

Reply:
Removed.